index.js

```diff
@@ -64,7 +64,7 @@ module.exports = (logFn, baseName) =>
                     })
                     .catch((err) => {
                         collected.measure[`${name}.error_duration`] = err.duration;
-                        throw err.reason;
+                        throw err.reason || err;
```

Comment:
can you provide me an example when the err won't have reason? I look through timed function and it seems pretty stable in that. Providing a `null` as a promise arg?

Reply:
Sure. We have (actually had, fixed in another PR) an issue that a non-promise object was provided as `promise` argument here. This caused an error like `[].then is not a function` being thrown that doesn't have a `reason` property.

Reply:
It could be a good idea but as an additional check (or maybe simply `Promise.resolve` given value).

Reply:
do whatever you find sensible